File: Leetcode/cp11_container_with_most_water/Solution.java
class Solution {
    public int maxArea(int[] height) {
        int max = 0;
        for (int i = 0; i < height.length; i++) {
          for (int j = 1; j < height.length; j++) {
	    # Finding column with lesser height from selected two columns
            int a = height[i];
            int b = height[j];
            int c = 0;
            if (a > b) {
              c = b;
            } else {
              c = a;
            }
	    # Finding width between selected two columns
            int d = j - i;
	    # Finding area of the possible container
            int tempRes = c * d;
            if (tempRes > max) {
              max = tempRes;
            }
          }
        }
        return max;
    }
}

File: Leetcode/cp11_container_with_most_water/Solution.py
class Solution:
    # Leetcode Problem: https://leetcode.com/problems/container-with-most-water/description/
    #
    # Time complexity: O(n)
    # Space complexity: O(1)
    def maxArea(self, height: list[int]) -> int:
        start = 0
        end = len(height) - 1
        max_area = 0
        while start < end:
            # Area that will hold the water will be:
            # height of min bar * space in between
            area = min(height[start], height[end]) * (end - start)
            max_area = max(max_area, area)
            # Move the pointer from the bar with min height
            if height[start] < height[end]:
                start += 1
            else:
                end -= 1
        return max_area
